File: FreeWaterTips.Utils/Environment.cs
using FreeWaterTips.Utils.Enums;

namespace FreeWaterTips.Utils
{
    public static class Environment
    {
        public static string Name => System.Environment.GetEnvironmentVariable("ENVIRONMENT") ?? String.Empty;

        public static EnvironmentEnum Type => Name switch
        {
            "Development" => EnvironmentEnum.Development,
            "Testing" => EnvironmentEnum.Testing,
            "Production" => EnvironmentEnum.Production,
            _ => EnvironmentEnum.Development
        };

        public static bool IsDevelopment => Type == EnvironmentEnum.Development;

        public static bool IsTesting => Type == EnvironmentEnum.Testing;

        public static bool IsProduction => Type == EnvironmentEnum.Production;

        public static string CosmosDBEndpoint => System.Environment.GetEnvironmentVariable("COSMOS_DB_ENDPOINT") ??
            throw new Exception("Environment variable 'COSMOS_DB_ENDPOINT' missing.");

        public static string CosmosDBKey => System.Environment.GetEnvironmentVariable("COSMOS_DB_KEY") ??
            throw new Exception("Environment variable 'COSMOS_DB_KEY' missing.");
    }
}

File: FreeWaterTips.API/src/utils/Environment.ts
import { EnvironmentEnum } from './EnvironmentEnum';

export class Environment {
  static get name(): string {
    return process.env.ENVIRONMENT || '';
  }

  static get type(): EnvironmentEnum {
    switch (this.name) {
      case 'Development':
        return EnvironmentEnum.Development;
      case 'Testing':
        return EnvironmentEnum.Testing;
      case 'Production':
        return EnvironmentEnum.Production;
      default:
        return EnvironmentEnum.Development;
    }
  }

  static get isDevelopment(): boolean {
    return this.type === EnvironmentEnum.Development;
  }

  static get isTesting(): boolean {
    return this.type === EnvironmentEnum.Testing;
  }

  static get isProduction(): boolean {
    return this.type === EnvironmentEnum.Production;
  }

  static get cosmosDBEndpoint(): string {
    const endpoint = process.env.COSMOS_DB_ENDPOINT;
    if (!endpoint) {
      throw new Error("Environment variable 'COSMOS_DB_ENDPOINT' missing.");
    }
    return endpoint;
  }

  static get cosmosDBKey(): string {
    const key = process.env.COSMOS_DB_KEY;
    if (!key) {
      throw new Error("Environment variable 'COSMOS_DB_KEY' missing.");
    }
    return key;
  }
}
